Simplenote/src/androidTest/java/com/automattic/simplenote/smoke/pages/BasePage.java

```diff
@@ -0,0 +1,312 @@
+
+    }
+
+    public static boolean isElementDisplayed(Integer elementID) {
+        return isElementDisplayed(visibleElementWithId(elementID));
+    }
+
+    public static boolean isElementDisplayed(ViewInteraction element) {
+        try {
+            element.check(matches(isDisplayed()));
+            return true;
+        } catch (Throwable e) {
+            return false;
+        }
+    }
+
+
+    public static ViewInteraction visibleElementWithId(Integer elementID) {
+        return onView(allOf(withId(elementID), withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)));
+    }
+
+    protected static Matcher<View> childAtPosition(
+            final Matcher<View> parentMatcher, final int position) {
+
+        return new TypeSafeMatcher<View>() {
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("Child at position " + position + " in parent ");
+                parentMatcher.describeTo(description);
+            }
+
+            @Override
+            public boolean matchesSafely(View view) {
+                ViewParent parent = view.getParent();
+                return parent instanceof ViewGroup && parentMatcher.matches(parent)
+                        && view.equals(((ViewGroup) parent).getChildAt(position));
+            }
+        };
+    }
+
+    protected Matcher<View> getElementFromMatchAtPosition(final Matcher<View> matcher, final int position) {
+        return new BaseMatcher<View>() {
+            int counter = 0;
+
+            @Override
+            public boolean matches(final Object item) {
+                if (matcher.matches(item)) {
+                    if (counter == position) {
+                        counter++;
+                        return true;
+                    }
+                    counter++;
+                }
+                return false;
+            }
+
+            @Override
+            public void describeTo(final Description description) {
+                description.appendText("Element at hierarchy position " + position);
+            }
+        };
+    }
+
+    /**
+     * Used for matrching colors but there is a problem related to sended color. Should be checked with source code.
+     *
+     * @param matcherColor
+     * @return
+     */
+    public static Matcher<View> textViewTextColorMatcher(final int matcherColor) {
+        return new BoundedMatcher<View, RobotoMediumTextView>(RobotoMediumTextView.class) {
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("with text color: " + matcherColor);
+            }
+
+            @Override
+            protected boolean matchesSafely(RobotoMediumTextView textView) {
+                return matcherColor == textView.getCurrentTextColor();
+            }
+        };
+    }
+
+    public static Matcher<View> hasTextColor(final int colorResId) {
```

Comment:
This method doesn't appear to be used anywhere in the project – does it need to be in this PR?

Reply:
I will try to be more careful about committing experimental codes.